### PR TITLE
Rework libretro support in CMake

### PR DESCRIPTION
**Pull request by Ghabry**
As this is not maintainable anymore I try to rewrite this now.
Marked as WIP, have to do many tests first.

Also updated libretro-common. Full of high quality "resync" commits.
https://github.com/libretro/libretro-common/commits/master

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,8 +13,15 @@ set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 
+# Need fPIC when compiling a shared library (e.g. libretro.so)
+# Must be at global scope, otherwise breaks -DPLAYER_BUILD_LIBLCF (see CMP0077)
+option(BUILD_SHARED_LIBS "Build shared easyrpg_libretro core" ON)
+if(BUILD_SHARED_LIBS)
+	set(CMAKE_POSITION_INDEPENDENT_CODE ON)
```
Comment (Ghabry) [inline CMakeLists.txt:20]: TODO: This also enables fPic/Pie for non-libretro builds, not really something we need.

**Comment by Ghabry** [comment]
@fmatthew5876 
can you check if this fixes the Visual STudio build for you?

**Comment by fmatthew5876** [comment]
After rebasing all my pending PRs and rerunning cmake I haven't hit the error yet. So I don't know if the error still exists.

I can confirm that I am able to build and link with this.

**Comment by Ghabry** [comment]
This results now in the following table which is what I wanted:
Is also not possible anymore that libretro fixes affect other builds because I operate now on a new ``easyrpg_libretro`` target instead of ``PROJECT_NAME``.

|  PLATFORM | SHARED_LIBS | Result              | fpic? |
|-----------|-------------|---------------------|-------|
|  SDL2     | ON          | easyrpg-player      | NO    |
| SDL2      | OFF         | easyrpg-player      | NO    |
| libretro  | ON          | easyrpg_libretro.so | YES   |
| libretro  | OFF         | easyrpg_libretro.a  | NO    |

**Review by fmatthew5876 — APPROVED** [review]

**Review by fdelapena — APPROVED** [review]

**Pull request merged**